project.go

```diff
@@ -3,8 +3,11 @@ package main
 import (
 	"bufio"
 	"bytes"
+	"context"
 	"fmt"
 	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
```

Comment:
Do we really need to add those dependencies just to implement a worker pool?

Reply:
It's not just a worker pool. We need a context as well to cancel all workers when an error has occurred. I think using these tested small dependencies outweighs having to write our custom implementation and maintaining it.

Reply:
Why do we need to cancel all the workers when an error occurred only in a single one?

Reply:
I suppose it depends on what the behavior should be like. But I simply kept the previous behavior, which is to stop walking the directory when `WalkTodosOfFile` returns an error.
Do you think we should handle this differently?

Reply:
@aoktayd is there any other solution that keeps the behavior reasonable (not necessarily the same), but avoids these dependencies?

Reply:
I've written a reasonable implementation (imo) without these dependencies https://github.com/tsoding/snitch/pull/166/commits/b6b0a099e5b4b503b553fc5f8decbdf62b2b92b4